src/library_exceptions.js

```diff
@@ -416,19 +416,13 @@ var LibraryExceptions = {
         // Catch all clause matched or exactly the same type is caught
         break;
       }
-      if ({{{ exportedAsmFunc('___cxa_can_catch') }}}(caughtType, thrownType, exceptionThrowBuf)) {
-        var adjusted = {{{ makeGetValue('exceptionThrowBuf', '0', '*') }}};
-        if (thrown !== adjusted) {
-          catchInfo.set_adjusted_ptr(adjusted);
-        }
+      if ({{{ exportedAsmFunc('___cxa_can_catch') }}}(caughtType, thrownType, catchInfo.get_adjusted_ptr_addr())) {
```

Comment:
Previously, the adjusted ptr is set only when we can catch it, but now it is always set. Is this fine?

Reply:
that is still true..  `__cxa_can_catch` only writes to its third argument if it returns non-zero:

https://github.com/emscripten-core/emscripten/blob/315d362ab124e9d2329aaa54f3ddb9c80b674f6c/system/lib/libcxxabi/src/private_typeinfo.cpp#L1346

Reply:
Ah right, I was confused.